### DISCUSSION TITLE
Have SEB server + Jitsi, how to use together?

### DISCUSSION THREAD

**Discussion by NTIPer** [SafeExamBrowser/seb-win-refactoring#219] [Q&A]
Hi, I just installed Jitsi server on a windows computer. Also I have installed Jitsi on another computer (ubuntu) for video calls. I want them connected to use with proctoring during live exams, like in this video: https://www.youtube.com/watch?v=6GEn_NWnZe8

Any tips how to do it? I can login to both servers and I can edit the settings in SEB server. But I cannot see a menu like in the youtube clip at 5:40.

**Comment by anhefti** [comment]
Hi, to set it up with SEB Server, please refer to the SEB Server User Guide at: https://seb-server.readthedocs.io/en/latest/exam_proctoring.html

**Reply by NTIPer** [reply]
Thank you, I tried using the guide but I am really unsure how to do. Right now I have the follow things to fix:
1. SEB server settings greyed out, cannot change them (but can change all other things in the server settings). See here: https://i.ibb.co/QbFtcHx/greryedout.jpg
2. When I try to add the Jitsi server it requires a key. I am not sure how to get this key from the Jitsi server (I have full access to the server). See here: https://i.ibb.co/1vWJ1DK/Key.jpg
3. I don't know how to make some configuration to send to a student. Do I do this in the normal SEB browser folder in the configuration tool and just make a .zip file that the student unzips? I do not have access to the students computers.

**Comment by anhefti** [comment]
1. It seems that the Exam Configuration is already in use by a running exam. Therefore the SEB Settings are in read-only mode for default. To change that you need to change the state of the Exam Configuration from "In Use" back to "Under Construction". To do so, please open the specified Exam Configuration in edit mode and change the state of the configuration as described. Save the Exam Configuration and go back to the SEB Settings. If you make changes to the SEB Settings please also make sure to Save/Publish them so the Exam has the newest SEB Settings available.
2. According to the Jitsi Meet Docker setup guide (https://jitsi.github.io/handbook/docs/devops-guide/devops-guide-docker), you need the value of the Jitsi Setting: "JWT_APP_ID" as Application Key and the value of Jitsi Setting: "JWT_APP_SECRET" as secret.
3. Please refer to the SEB Server User guide for this (https://seb-server.readthedocs.io/en/latest/connection_config.html). You need a Connection Configuration that you can create in SEB Server. You can then download this Connection Configuration and send it to students or make it available within a download link.

**Reply by NTIPer** [reply]
Thank you so much for the help!!

1. Solved, it was like you said.
2. I followed this guide: https://jitsi.github.io/handbook/docs/devops-guide/devops-guide-quickstart 
I can't even find the config file for jitsi. Not used to linux/ubuntu at all. If I can find the config file I could change the values for JWT_APP_ID and so on. Any tips here?
3. I made a Connection Configuration and downloaded it. When I try running it I get an error message after typing the password: https://i.ibb.co/1XxHGZG/SEBServer-Settings-seb2.jpg

I know these are probably basic questions, like where to find the config file for jitsi, and jitsi is not even part of SEB. But if you have time to answer again I would really appreciate it.

**Comment by anhefti** [comment]
1. perfect!
2. sorry, currently not. What instance of Jitsi Meet have you running?
3. maybe your SEB Server setup is not correct. Have you correctly set the following SEB Server application.properties: 

```
\# Set the below URL component settings from where the the web-service is externally available
\# NOTE: This must be the address from that the service is externally available (not network inernal address)
- sebserver_webservice_http_external_scheme=
- sebserver_webservice_http_external_servername=
- sebserver_webservice_http_external_port=
```

You can also see if this is set correctly in the SEB Server logs on startup. Please make sure that the following, red marked URL's from the SEB Server log are pointing to the URL of the SEB Server where it is accessible from the Internet:
![image](https://user-images.githubusercontent.com/39078777/136196553-3dfa4144-79fa-4919-8d52-462021ff2f67.png)

If this is not the case, you need to set the correct properties values above. If you need further help with that, we need the information what SEB Server setup you use.

**Reply by NTIPer** [reply]
2. I am using the newest version of Jitsi-meet. I got prosody like this `http://packages.prosody.im/debian $(lsb_release -sc)` and I got Jitsi like this `sudo apt install jitsi-meet` and got it yesterday/today.  

3. It should be correct I think. I have opened the port also in my router. Right now I am just testing, and if it works out as good as it seems to be we will hire a server somewhere. So right now I have full control of router and the SEB server + Jitsi server locally at home for testing.
I just chose a severname, maybe it matters what I choose there? I can connect to the SEB sever and change settings through the link with IP below so that one works. 
sebserver.webservice.http.external.scheme=http://92.32.89.38:8080/
sebserver.webservice.http.external.servername=hemma
sebserver.webservice.http.external.port=8080

**Comment by anhefti** [comment]
2. Sorry I don't have any experiences with this Jitsi Meets service. Maybe you can search the forums on how you can get the Jtisi Meet settings and configurations for that setup.
3. The settings you posted are slightly incorrect because **sebserver.webservice.http.external.scheme** should only define the scheme of the URL (http or https). While the URL **http://hemma:8080** seems not to work, the URL **http://92.32.89.38:8080/** is working. So according to that please set the settings as follows:

sebserver.webservice.http.external.scheme=http
sebserver.webservice.http.external.servername=92.32.89.38
sebserver.webservice.http.external.port=8080

Then it should work.

**Reply by NTIPer** [reply]
3. I see now and changed as you wrote. But I still get the same error message from SEB as before:
https://i.ibb.co/1XxHGZG/SEBServer-Settings-seb2.jpg

It doesn't say much about what might be wrong. I can still connect to the SEB server and change settings as normal. Do you have any tips here, or can I send some files or something to look at?

**Reply by brucenelm** [reply]
You are using the Demo SebServer. The .seb settings you download point to the demo SEB-Server that belongs to Zurich. Could you edit it and point it to your SEB-Server URL?
